libraries/src/Helper/MediaHelper.php

```diff
@@ -462,4 +471,88 @@ public static function isValidLocalDirectory($directory)
 
 		return false;
 	}
+
+	/**
+	 * Method to generate different-sized versions of a form image
+	 *
+	 * @param   string  $initImage   Image value before submission
+	 * @param   string  $finalImage  Image value during submission
+	 * @param   mixed  	$sizes       String or array of strings. Example: $sizes = array('1200x800','800x600');
+	 *
+	 * @return  void
+	 *
+	 * @since   4.1.0
+	 */
+	public static function generateResponsiveFormImages($initImage, $finalImage, $sizes = null)
+	{
+		// Use default if sizes are not provided
+		if(is_null($sizes))
+		{
+			$sizes = static::$responsiveSizes;
+		}
+
+		// Remove previously generated images if original is changed
+		if($initImage !== "" && $initImage !== $finalImage)
+		{
+			$imgObj = new Image(JPATH_ROOT . '/' . $initImage);
+			$imgObj->deleteMultipleSizes();
+		}
+
+		// Generate new responsive images if file exists
+		if(is_file(JPATH_ROOT . '/' . $finalImage))
+		{
+			$imgObj = new Image(JPATH_ROOT . '/' . $finalImage);
+			$imgObj->createMultipleSizes($sizes);
+		}
+	}
+
+	/**
+	 * Method to generate different-sized versions of content images
+	 *
+	 * @param   string  $initContent   Content before submission
+	 * @param   string  $finalContent  Content during submission
+	 * @param   mixed   $sizes         String or array of strings. Example: $sizes = array('1200x800','800x600');
+	 *
+	 * @return  void
+	 *
+	 * @since   4.1.0
+	 */
+	public static function generateResponsiveContentImages($initContent, $finalContent, $sizes = null)
+	{
+		// Use default if sizes are not provided
+		if(is_null($sizes))
+		{
+			$sizes = static::$responsiveSizes;
+		}
+
+		$pattern = '/src="([^"]+)"/';
```

Comment:
What happens if someone provides $initContent that sort of content, that contains other tags with a src attribute? F.e. audio, video, embed, input, iframe (...).  

have a look @$data['articletext']